File: Sprint8/D/C++/main.cpp
// https://contest.yandex.ru/contest/26131/problems/D/
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

int main() {
    int n;
    std::cin >> n;

    std::vector<std::string> strs(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> strs[i];
    }

    if (n == 1) {
        std::cout << strs[0].length() << std::endl;
        return 0;
    }

    std::sort(strs.begin(), strs.end());

    std::string& first = strs[0];
    std::string& last = strs[n - 1];
    int i = 0;

    while (i < first.length() && i < last.length() && first[i] == last[i]) {
        i++;
    }

    std::cout << i << std::endl;
}

File: Sprint8/D/Rust/src/main.rs
// https://contest.yandex.ru/contest/26131/problems/D/
use std::io::BufRead;

fn common_prefix_length(strs: &[String]) -> usize {
    if strs.len() == 1 {
        return strs[0].len();
    }

    let (first, last) = (strs.first().unwrap(), strs.last().unwrap());
    first
        .chars()
        .zip(last.chars())
        .take_while(|(a, b)| a == b)
        .count()
}

fn main() {
    let mut lines = std::io::stdin().lock().lines();
    let n: usize = lines.next().unwrap().unwrap().parse().unwrap();

    let mut strs = Vec::with_capacity(n);
    for _ in 0..n {
        strs.push(lines.next().unwrap().unwrap());
    }

    println!("{}", common_prefix_length(&strs));
}
